testing/e2e/e2e_api_test.go

```diff
@@ -515,9 +515,228 @@ func (s *BeaconKitE2ESuite) TestGetStateValidatorByPubkey() {
 
 // TestGetStateValidatorInvalidID tests getting the state validator with an invalid id.
 func (s *BeaconKitE2ESuite) TestGetStateValidatorInvalidID() {
-	resp, err := s.getStateValidator("head", "invalid_id")
+	resp, err := s.getStateValidator(utils.StateIDHead, "invalid_id")
 	s.Require().NoError(err)
 	s.Require().NotNil(resp, "response should not be nil")
 	s.Require().Equal(http.StatusBadRequest, resp.StatusCode)
 	defer resp.Body.Close()
 }
+
+// getValidatorBalances gets the validator balances for the given stateID and optional ids.
+func (s *BeaconKitE2ESuite) getValidatorBalances(stateID string, ids ...string) (*http.Response, error) {
+	client := s.initHTTPBeaconTest()
+
+	url := fmt.Sprintf("/eth/v1/beacon/states/%s/validator_balances", stateID)
+
+	// Add ID parameters if provided
+	if len(ids) > 0 {
+		queryParams := make([]string, 0, len(ids))
+		for _, id := range ids {
+			queryParams = append(queryParams, "id="+id)
+		}
+		url = url + "?" + strings.Join(queryParams, "&")
+	}
+
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get validator balances: %w", err)
+	}
+	if resp == nil {
+		return nil, errors.New("received nil response")
+	}
```

Comment:
this is checked on line 585 as well? Consider handling nil responses only in one place

Reply:
Wow! Kept it in func `getValidatorBalances` and removed NotNil check.